src/bootstrap/src/bin/rustc.rs

```diff
@@ -145,6 +145,35 @@ fn main() {
         cmd.arg("-Z").arg("on-broken-pipe=kill");
     }
 
+    // In `cargo.rs` we add the `FLAGS_FOR_RUSTC` flags for all `compiler/`
+    // crates to RUSTFLAGS. However, RUSTFLAGS is ignored for `compiler/` proc
+    // macro crates, because RUSTFLAGS is ignored when `--target` is given. So,
```

Comment:
> RUSTFLAGS is ignored when `--target` is given

That's not quite true, it is not ignored. But it only applies to target crates, not to host crates.